Source/WebKit/WebProcess/WebPage/RemoteLayerTree/GraphicsLayerCARemote.cpp

```diff
@@ -90,4 +93,62 @@ void GraphicsLayerCARemote::moveToContext(RemoteLayerTreeContext& context)
     context.graphicsLayerDidEnterContext(*this);
 }
 
+class GraphicsLayerCARemoteAsyncContentsDisplayDelegate : public GraphicsLayerAsyncContentsDisplayDelegate {
+public:
+    GraphicsLayerCARemoteAsyncContentsDisplayDelegate(IPC::Connection& connection, DrawingAreaIdentifier identifier, WebCore::GraphicsLayer::PlatformLayerID layerID)
+        : m_connection(connection)
+        , m_drawingArea(identifier)
+        , m_layerID(layerID)
+    { }
+
+    bool hasShareableBackend(ImageBuffer& buffer)
+    {
+        auto* backend = buffer.ensureBackendCreated();
+        if (!backend)
+            return false;
+
+        buffer.flushDrawingContext();
+
+        auto* sharing = backend->toBackendSharing();
+        return is<ImageBufferBackendHandleSharing>(sharing);
+    }
+
+    bool tryCopyToLayer(ImageBuffer& buffer) final
+    {
+        if (!hasShareableBackend(buffer))
+            return false;
+        auto clone = buffer.clone();
+        if (!clone)
+            return false;
+        ASSERT(hasShareableBackend(*clone));
+        auto* backend = clone->ensureBackendCreated();
+        if (!backend)
+            return false;
+
+        clone->flushDrawingContext();
+
+        auto* sharing = backend->toBackendSharing();
+        if (!is<ImageBufferBackendHandleSharing>(sharing))
```

Comment:
This is what `hasShareableBackend()` did. Maybe just don't bother checking `hasShareableBackend()` at the top of the function, and just clone and check here.

Reply:
The idea was to try check if the original buffer could be shareable, before committing to the cost of cloning (and then verifying that the clone is also shareable).